service_kit/logging/git.py

```diff
@@ -0,0 +1,118 @@
+import shutil
+import subprocess
+from pathlib import Path
+
+import yaml
+
+from . import logger
+
+GIT = shutil.which("git")
+
+
+def log_git_status(git_status_yaml_path: Path | None = None):
+    """
+    Log the status of the current git repository
+
+    Logs the commit ID, repository status (clean or dirty), and any tags associated with the current
+    HEAD at the INFO level. Optionally, this function accepts a path to a YAML file. If, and only
+    if, the current directory is not a git repository, or the `git` command is not available, the
+    status can be read from the YAML file instead. Note, however, that this function cannot
+    guarantee the accuracy of the information in the YAML file.
+
+    Example YAML file:
+        commit: "3db2b4fbc9db2635b4ae6411496132f6d985426e"
+        status: "clean"
+        tags:
+        - test-tag
+
+    :param git_status_yaml_path: Optional path to a YAML file containing git status information.
+    """
+    if GIT is None or not _pwd_is_git_repository():
+        logger.warning(
```

Comment:
Ideal case would be to log if git doesn't exist or if it's not a git repository, but if we don't we should log that git not found OR project is not a git repository